### PR TITLE
[Supplier] Add `min_stake` supplier module param

### PR DESCRIPTION
**Pull request by bryanchriswhite**
## Summary

- Adds the `min_stake` param to the supplier module following section 0 from the updated docs (see https://github.com/pokt-network/poktroll/pull/839).
- Adds missing validation unit test coverage around existing application module param, max_delegated_gateways.
- Pushes validation from the MsgUpdateParam handler down into MsgUpdateParam#ValidateBasic().
- Ensures Params#Valid() is called prior to setting params in MsgUpdateParam handler.
- Updates error returns in the same handler to ensure all error paths return appropriate gRPC status errors.

## Dependencies

- #809
- #843 
- #844 
- #845
- #847
- #848
- #849

## Dependents
  
- #857
- #852 
- #861 
- #851 
- #863 

## Issue

- #612

## Type of change

Select one or more from the following:

- [x] New feature, functionality or library
- [ ] Consensus breaking; add the `consensus-breaking` label if so. See #791 for details
- [ ] Bug fix
- [ ] Code health or cleanup
- [ ] Documentation
- [ ] Other (specify)

## Testing

- [ ] **Documentation**: `make docusaurus_start`; only needed if you make doc changes
- [ ] **Unit Tests**: `make go_develop_and_test`
- [ ] **LocalNet E2E Tests**: `make test_e2e`
- [ ] **DevNet E2E Tests**: Add the `devnet-test-e2e` label to the PR.

## Sanity Checklist

- [x] I have tested my changes using the available tooling
- [x] I have commented my code
- [x] I have performed a self-review of my own code; both comments & source code
- [ ] I create and reference any new tickets, if applicable
- [x] I have left TODOs throughout the codebase, if applicable

### CODE REVIEW

File: makefiles/params.mk [hunk 1]
```diff
@@ -127,6 +127,19 @@ params_update_application_max_delegated_gateways: ## Update the application modu
 params_update_application_min_stake: ## Update the application module min_stake param
 	poktrolld tx authz exec ./tools/scripts/params/application_min_stake.json $(PARAM_FLAGS)
 
+### Gateway Module Params ###
```
Comment (bryanchriswhite) [inline makefiles/params.mk:130]:
```suggestion
### Supplier Module Params ###
```

File: x/supplier/keeper/msg_server_update_param.go [hunk 2]
```diff
@@ -3,35 +3,47 @@ package keeper
...
 	switch msg.Name {
+	case suppliertypes.ParamMinStake:
+		params.MinStake = msg.GetAsCoin()
 	default:
-		return nil, suppliertypes.ErrSupplierParamInvalid.Wrapf("unsupported param %q", msg.Name)
+		return nil, status.Error(
+			codes.InvalidArgument,
+			suppliertypes.ErrSupplierParamInvalid.Wrapf("unsupported param %q", msg.Name).Error(),
+		)
+	}
```
Comment (bryanchriswhite) [inline x/supplier/keeper/msg_server_update_param.go:39]:
```suggestion
	}
	
	if err := params.Validate(); err != nil {
		return nil, status.Error(codes.InvalidArgument, err.Error())	
	}
```

_TODO: Also update in "adding_params.md" doc._

File: x/supplier/keeper/msg_server_stake_supplier.go [hunk 3]
```diff
@@ -99,22 +106,23 @@ func (k msgServer) StakeSupplier(ctx context.Context, msg *types.MsgStakeSupplie
...
 	// Retrieve the account address of the message signer
 	msgSignerAddress, err := sdk.AccAddressFromBech32(msg.Signer)
 	if err != nil {
-		logger.Error(fmt.Sprintf("could not parse address %q", msg.Signer))
-		return nil, err
+		logger.Info(fmt.Sprintf("ERROR: could not parse address %q", msg.Signer))
+		return nil, status.Error(codes.InvalidArgument, err.Error())
 	}
 
 	// Send the coins from the message signer account to the staked supplier pool
```
Comment (red-0ne) [inline x/supplier/keeper/msg_server_stake_supplier.go:121]: Shouldn't we enforce that the new stake is always greater than min stake?

**Review by red-0ne — COMMENTED** [review]
Left a comment about enforcing min stake when staking, otherwise LGTM.

**Pull request merged**